source/common/http/conn_manager_impl.cc

```diff
@@ -913,6 +935,18 @@ void ConnectionManagerImpl::ActiveStream::maybeEndDecode(bool end_stream) {
   }
 }
 
+void ConnectionManagerImpl::ActiveStream::disarmRequestTimer() {
+  if (request_timer_) {
+    request_timer_->disableTimer();
+  }
+}
+
+void ConnectionManagerImpl::ActiveStream::maybeDisarmRequestTimer() {
+  if (request_info_.lastUpstreamTxByteSent().has_value()) {
+    disarmRequestTimer();
```

Comment:
From Alyssa's comment in the issue, it sounds like this timeout is intended to be for the whole request/response pathway (all of the filters encoding, and all of their decoding).  So shouldn't we not stop the timeout until the response has been sent back to the consumer?

(I also threw a comment on the issue to clarify this.)